src/ngraph/pass/algebraic_simplification.cpp

```diff
@@ -55,6 +59,82 @@ static std::shared_ptr<pattern::op::Label>
     return std::dynamic_pointer_cast<pattern::op::Label>(matcher->pattern_node()->get_argument(1));
 }
 
+//`simplify_concat` identifies slices-concat sequences
+// that cancel each other. Namely it replaces subgraphs
+//similar to the one below with `arg`
+//
+//                 +----------+
+//            +----+slice(n/2..n)---+
+// +-------+  |    +----------+    |  +-----------+
+// |  arg  +--+                    +--+  concat   |
+// +-------+  |    +----------+    |  +-----------+
+//            +----+slice(0..n/2)---+
+//                 +----------+
+static bool simplify_concat(std::shared_ptr<Node> n)
+{
+    NGRAPH_DEBUG << "In simplify_concat for " << n->get_name();
+
+    std::shared_ptr<Node> goe;
+
+    auto lgoe = std::make_shared<pattern::op::Label>(element::i32, Shape{2, 1});
+
+    auto slice =
+        std::make_shared<op::Slice>(lgoe, Coordinate{0, 0}, Coordinate{2, 1}, Strides{1, 1});
+
+    auto reshape_pred = [](std::shared_ptr<Node> r) {
+        return std::dynamic_pointer_cast<op::Reshape>(r) != nullptr;
+    };
+
+    auto skip_reshape = std::make_shared<pattern::op::Skip>(slice, reshape_pred);
+
```

Comment:
Can we have this without reshape predicate?, it will cause a layout exception if we have case like this  `GOE(32, 200) -> Slice( 16, 200) -> Reshape(16, 1, 200) -> Result(16, 1, 200)` this pass, will reduce it to `GOE(32, 200) -> Result(16, 1, 200)`. which will cause layout exception.